### PR TITLE
Integration Testing

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

File: src/CalldataList.sol [hunk 1]
```diff
@@ -68,40 +61,33 @@ abstract contract CalldataList {
...
 
-        Index[] storage calldataChecks =
-            calldataList[contractAddress][selector].calldataChecks;
+        Index[] storage calldataChecks = calldataList[contractAddress][selector];
 
         require(
             calldataChecks.length > 0, "CalldataList: No calldata checks found"
         );
-        require(
-            value <= calldataList[contractAddress][selector].maxValue,
```
Comment (prateek105) [inline src/CalldataList.sol:94]: shouldn't there be a cap on value that can be sent in a transaction. If some users want no max value cap, they can set `maxValue` to `type(uint256).max`

File: test/integration/System.t.sol [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+            /// each morpho blue function call needs two checks:
+            /// 1). check the pool id where funds are being deposited is whitelisted.
+            /// 2). check the recipient of the funds is whitelisted whether withdrawing
+            /// or depositing.
+
+            uint16[] memory startIndexes = new uint16[](8);
+            /// morpho blue supply
+            startIndexes[0] = 4;
+            startIndexes[1] = 4 + 32 * 7 + 12;
+            /// only grab last twenty bytes of the 7th argument
```
Comment (prateek105) [inline test/integration/System.t.sol:443]: move comments above the loc.

File: test/integration/System.t.sol [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+
+        calldatas[1] = abi.encodeWithSelector(
+            GuardManager.setGuard.selector, address(restricted)
+        );
+
+        calldatas[2] = abi.encodeWithSelector(
+            ModuleManager.enableModule.selector, address(timelock)
+        );
+
+        call3.Call3[] memory calls3 = new call3.Call3[](3);
```
Comment (anajuliabit) [inline test/integration/System.t.sol:294]: If we move this declaration to the first line of the function, we won't need to declare the temporary variables above like `calls` and `calldatas`.

File: test/integration/System.t.sol [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+            startIndexes[7] = 4 + 32 * 8 + 12;
+            /// only grab last twenty bytes of the 8th argument
+
+            uint16[] memory endIndexes = new uint16[](8);
+            /// morpho blue supply
+            endIndexes[0] = 4 + 32 * 5;
+            endIndexes[1] = 4 + 32 * 7 + 32;
+            /// last twenty bytes represents who supplying on behalf of
+            /// ethena usd approve morpho
+            endIndexes[2] = 36;
```
Comment (anajuliabit) [inline test/integration/System.t.sol:464]:
```suggestion
            endIndexes[0] = startIndexes[0] + 32 * 5;
            endIndexes[1] = startIndexes[1] + 20;
            /// last twenty bytes represents who supplying on behalf of
            /// ethena usd approve morpho
            endIndexes[2] =  startIndexes[2] + 20;
```

File: test/integration/System.t.sol [hunk 5]
```diff
@@ -0,0 +1,632 @@
...
+            bytes memory sig1 = abi.encodePacked(r1, s1, v1);
+            bytes memory sig2 = abi.encodePacked(r2, s2, v2);
+            bytes memory sig3 = abi.encodePacked(r3, s3, v3);
+
+            collatedSignatures = abi.encodePacked(sig1, sig2, sig3);
+        }
+
+        safe.checkNSignatures(
+            transactionHash, innerCalldatas, collatedSignatures, 3
+        );
```
Comment (anajuliabit) [inline test/integration/System.t.sol:569]: Llet's add a function for this code as it is being repeated in different tests

File: test/integration/System.t.sol [hunk 6]
```diff
@@ -0,0 +1,632 @@
...
+            address(timelock),
+            ""
+        );
+
+        IMorphoBase(morphoBlue).accrueInterest(
+            MarketParams(dai, ethenaUsd, oracle, irm, lltv)
+        );
+
+        vm.prank(owners[0]);
+        timelock.executeWhitelistedBatch(targets, values, calldatas);
```
Comment (anajuliabit) [inline test/integration/System.t.sol:630]: let's check morploBlue state after calling execute to make sure it worked

**Review by anajuliabit — CHANGES_REQUESTED** [review]

File: test/integration/System.t.sol [hunk 7]
```diff
@@ -628,5 +604,48 @@ contract SystemIntegrationTest is Test {
...
+        returns (bytes32 marketParamsId)
+    {
+        assembly ("memory-safe") {
+            marketParamsId :=
+                keccak256(marketParams, MARKET_PARAMS_BYTES_LENGTH)
+        }
+    }
+
+    function signTxAllOwners(bytes32 transactionHash)
+        public
```
Comment (anajuliabit) [inline test/integration/System.t.sol:637]: If we intend to use these helper functions across other test files I believe we should create a base contract to this  and inherit here

If these helpers are only useful for integration tests, then I think we should make it private

File: test/integration/System.t.sol [hunk 8]
```diff
@@ -0,0 +1,632 @@
...
+    /// keccak256("guard_manager.guard.address")
+    uint256 private constant GUARD_STORAGE_SLOT =
+        0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8;
+
+    address[] public owners;
+
+    uint256 public startTimestamp;
+
+    function setUp() public {
+        startTimestamp = block.timestamp;
```
[the remark this answers is not in the document]
Comment (anajuliabit) [reply test/integration/System.t.sol:143]: so let delete the comment then

File: test/integration/System.t.sol [hunk 9]
```diff
@@ -0,0 +1,651 @@
...
+        // at least start at unix timestamp of 1m so that block timestamp isn't 0
```
Comment (anajuliabit) [inline test/integration/System.t.sol:144]:
```suggestion
```

**Review by anajuliabit — APPROVED** [review]
lgtm

**Pull request merged**